src/D2L.Bmx/VirtualTerminal.cs

```diff
@@ -0,0 +1,54 @@
+using System.Runtime.InteropServices;
+
+namespace D2L.Bmx;
+
+// The code here exists because legacy Windows console doesn't support ANSI escape codes by default.
+// We can consider deleting everything here once Windows 10 is out of support and
+// Windows Terminal becomes the default console on all supported Windows versions.
+internal static partial class VirtualTerminal {
+	private const string Kernel32 = "kernel32.dll";
+	private const int STD_ERROR_HANDLE = -12;
+	private const uint ENABLE_VIRTUAL_TERMINAL_PROCESSING = 0x0004;
+
+	private static bool _enablingAttempted = false;
+	private static bool _enabled = false;
+
+	// https://learn.microsoft.com/en-us/windows/console/classic-vs-vt
+	public static bool TryEnableOnStderr() {
+		if( _enablingAttempted ) {
+			return _enabled;
+		}
+		_enablingAttempted = true;
+
+		if( !RuntimeInformation.IsOSPlatform( OSPlatform.Windows ) ) {
+			// POSIX systems all support ANSI escape codes
+			_enabled = true;
+			return true;
+		}
+		nint stderrHandle = GetStdHandle( STD_ERROR_HANDLE );
+		if( !GetConsoleMode( stderrHandle, out int mode ) ) {
+			_enabled = false;
+			return false;
+		}
+		if( ( mode & ENABLE_VIRTUAL_TERMINAL_PROCESSING ) == ENABLE_VIRTUAL_TERMINAL_PROCESSING ) {
+			_enabled = true;
+			return true;
+		}
+		_enabled = SetConsoleMode( stderrHandle, (int)( mode | ENABLE_VIRTUAL_TERMINAL_PROCESSING ) );
+		return _enabled;
+	}
+
+	// https://learn.microsoft.com/en-us/windows/console/getstdhandle
+	[LibraryImport( Kernel32 )]
+	private static partial IntPtr GetStdHandle( int nStdHandle );
```

Comment:
You sure we don't need to do [``SetLastError = true``](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.dllimportattribute.setlasterror)  here?

If we are calling into un-managed code & checking if that was a success (I might be mis-understanding the purpose of that flag).

``On .NET, the error information is cleared (set to 0) before invoking the callee when this field is set to true``
So maybe that only applies for repeated calls with subsequent successes?